src/pyodide-py/pyodide/_base.py

```diff
@@ -42,6 +43,31 @@ def open_url(url: str) -> StringIO:
     return StringIO(req.response)
 
 
+def quiet_code(code: str) -> bool:
+    """
+    Does the last nonwhitespace character of code is a semicolon?
+
+    This can be overridden to customize the way eval_code is silenced.
+    """
+    # largely inspired from IPython:
+    # https://github.com/ipython/ipython/blob/86d24741188b0cedd78ab080d498e775ed0e5272/IPython/core/displayhook.py#L84
+
+    codeio = StringIO(code)
+    tokens = list(tokenize.generate_tokens(codeio.readline))
+
+    for token in reversed(tokens):
+        if token.type in (
+            tokenize.ENDMARKER,
+            tokenize.NL,
+            tokenize.NEWLINE,
```

Comment:
It looks like `NL` starts each line and `NEWLINE` ends the line? Do you need to skip both in case the line is completely empty?

Reply:
It depends on what we want. To me, evaluating `"1+1;\n\n"` should return `None` and `1+1\n\n` should return `2`. In that case, `NL` should be skipped. WDYT?

Reply:
I'll add tests for that.

Reply:
> To me, evaluating `"1+1;\n\n"` should return `None` and `1+1\n\n` should return `2`. In that case, `NL` should be skipped.